cv-data-controller/src/main/java/com/trihydro/cvdatacontroller/controller/CdotUpstreamPathController.java

```diff
@@ -27,366 +25,511 @@
+            logger.debug("Geojson string for buffer: {}", geojsonString);
+        }
+        return ResponseEntity.ok(buffer);
     }
-  }
-
-  /**
-   * Context class for traversing mileposts to get buffer path
-   */
-  public static class TraverseContext {
-    private final List<Milepost> allMileposts;
-    private final int startIndex;
-    private final double desiredDistanceInMiles;
-    private final PathDirection direction;
-
-    private TraverseStrategy traverseStrategy;
-
-    private List<Milepost> buffer;
-    private double distanceInMiles;
-
-    public TraverseContext(List<Milepost> allMileposts, int startIndex,
-                           double desiredDistanceInMiles, PathDirection direction) {
-      this.allMileposts = allMileposts;
-      this.startIndex = startIndex;
-      this.desiredDistanceInMiles = desiredDistanceInMiles;
-      this.direction = direction;
-      this.buffer = new ArrayList<>();
+
+    /**
+     * Retrieves all mileposts for a given route from the CDOT GIS service.
+     *
+     * <p>This method uses `CdotGisService.getRouteById()` to retrieve the route information
+     * in JSON format. The JSON response contains all the latitude and longitude points
+     * in the route. This information is then extracted into the `Milepost` model and
+     * returned as a list of mileposts.</p>
+     *
+     * @param routeId the ID of the route to retrieve mileposts for
+     * @return a list of `Milepost` objects representing the mileposts in the route
+     * @throws JsonProcessingException if there is an error processing the JSON response
+     * @throws RestClientException     if an error occurs while making the request
+     */
+    public List<Milepost> getMilepostsForRoute(String routeId) throws JsonProcessingException,
+            RestClientException {
+        ResponseEntity<String> response = cdotGisService.getRouteById(routeId);
+        return getMilepostsFromResponse(response, routeId);
     }
 
-    public void performTraversal() {
-      traverseStrategy.traverse(this);
+    /**
+     * Determines if the route is supported.
+     *
+     * <p>This method uses `CdotGisService.getRouteById()` to retrieve the route information
+     * in JSON format. If the JSON format does not contain the route, it is not supported.</p>
+     *
+     * @param routeId the ID of the route to retrieve mileposts for
+     * @return boolean indicating whether the route is supported or not
+     */
+    @RequestMapping(method = RequestMethod.GET, produces = "application/json", value = "/get-route-supported/{routeId}")
+    public ResponseEntity<Boolean> isRouteSupported(@PathVariable String routeId) {
+        ResponseEntity<String> response = cdotGisService.getRouteById(routeId);
+        try {
+            String routeJsonString = response.getBody();
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode rootNode = objectMapper.readTree(routeJsonString);
```

Comment:
Suggestion (non-blocking): Deserialize this JSON into a dedicated POJO representing the data returned from the GIS endpoint. This will make the code less brittle and easier to read.